### PR TITLE
Reduce memory usage in skymatch

### PR DESCRIPTION
**Pull request by mcara**
**Description**

This PR reduces memory usage when running `skymatch` step when input to the step is an ASN table. This is accomplished by saving large image and mask arrays to temporary files and retrieving them on demand. This PR will have no effect when input to the step is a list of already loaded into memory data models.

NOTE: Modifications to `jwst/datamodels/container.py` were designed by @stsci-hack who provided the file to me. I made minor adjustments, I think to `models_grouped`.

Checklist
- [ ] Tests
- [x] Documentation
- [x] Change log
- [x] Milestone
- [x] Label(s)

### CODE REVIEW

File: jwst/datamodels/container.py [hunk 1]
```diff
@@ -445,6 +467,77 @@ def ind_asn_type(self, asn_exptype):
...
+            data_item_size = model.data.itemsize
+            data_item_type = model.data.dtype
+            model.close()
+        del model
+
+        section_nrows = min(imrows, int(self.buffer_size / (imcols * data_item_size)))
+
+        if section_nrows == 0:
+            self.buffer_size = imcols * data_item_size
+            print("WARNING: Buffer size is too small to hold a single row.\n"
```
Comment (hbushouse) [inline jwst/datamodels/container.py:499]: Don't want print statements in production code. Change to log message.

**Review by hbushouse — COMMENTED** [review]
Looks OK to me. Just one trivial comment. Would be good to have @stsci-hack review for compatibility with his work on changing datamodels for use in outlier_detection.

File: jwst/datamodels/container.py (hunk 1 above)
Comment (mcara) [reply jwst/datamodels/container.py:499]: Looking at the code for this function, I think it needs serious revision. I'll do my best but @stsci-hack needs to review this. In principle he can fix the code later when working on `outlier_detection`. I am not using this part of the code.

Comment (mcara) [reply jwst/datamodels/container.py]: @hbushouse @stsci-hack done (fixed `set_buffer` and `print()`->logging).

**Review by stsci-hack — COMMENTED** [review]
I agree that print statements should not be used and support changing those calls to logging calls in this version of the updated module.  These changes can be ported over to the outlier-detection update as soon as the code meets the needs of this PR.

**Pull request merged**